migrations/00004_notifications/index.sql

```diff
@@ -0,0 +1,114 @@
+DROP TABLE IF EXISTS notification_types,
+    notification_subscriptions,
+    notification_mediums,
+    notification_medium_configurations CASCADE;
+
+--------------------------------------------
+-- Notification types, e.g.INCOMING_TOKEN --
+--------------------------------------------
+CREATE TABLE notification_types(
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL
+);
+
+-- TODO: Confirm these types
+INSERT INTO notification_types (name) VALUES
+    ('DELETED_MULTISIG_TRANSACTION'),
+    ('EXECUTED_MULTISIG_TRANSACTION'),
+    ('INCOMING_ETHER'),
+    ('INCOMING_TOKEN'),
+    ('MESSAGE_CREATED'),
+    ('MODULE_TRANSACTION'),
+    ('NEW_CONFIRMATION'),
+    ('MESSAGE_CONFIRMATION'),
+    ('OUTGOING_ETHER'),
+    ('OUTGOING_TOKEN'),
+    ('PENDING_MULTISIG_TRANSACTION'),
+    ('SAFE_CREATED');
+
+----------------------------------------------------------------------
+-- Chain-specific Safe notification preferences for a given account --
+----------------------------------------------------------------------
+CREATE TABLE notification_subscriptions(
+    id SERIAL PRIMARY KEY,
+    account_id INT NOT NULL,
```

Comment:
We likely need to include some form of device ID (like the current implementation) as well as the push token can differ across devices.